### PR TITLE
Sprint 3 Node Clean-up

### PR DESCRIPTION
**Pull request by bencloward**
Improved the layout of several of the new subgraphs
Improved the performance of the InteriorCubemap node.
Removed "Node" from the name of the Gradient node and removed the label from the gradient
Added a couple of test shaders.
Added Boolean, Integer, Vec2, Vec3, and Vec4 nodes.
Added missing semicolon to the FadeTransition node
Fixed a typo in the power node

### CODE REVIEW

File: com.unity.shadergraph/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/Input/Basic/IntegerNode.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+            "Integer", // Name
+            "Out = IntConst;",
+            new ParameterDescriptor("IntConst", TYPE.Int, Usage.Static),
+            new ParameterDescriptor("Out", TYPE.Int, Usage.Out)
+        );
+
+        public static Dictionary<string, string> UIStrings => new()
+        {
+            { "Category", "Input, Basic" },
+            { "Name.Synonyms", "whole number" },
```
Comment (tracychen1234) [inline com.unity.shadergraph/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/Input/Basic/IntegerNode.cs:20]: maybe also "vec1"?

File: com.unity.shadergraph/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/Artistic/Filter/FadeTransitionNode.cs [hunk 2]
```diff
@@ -9,7 +9,7 @@ internal class FadeTransitionNode : IStandardNode
         public static FunctionDescriptor FunctionDescriptor => new(
             1,
             "FadeTransition",
-            "Fade = saturate(FadeValue*(FadeContrast+1)+(NoiseValue-1)*FadeContrast)",
+            "Fade = saturate(FadeValue*(FadeContrast+1)+(NoiseValue-1)*FadeContrast);",
```
Comment (tracychen1234) [inline com.unity.shadergraph/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/Artistic/Filter/FadeTransitionNode.cs:12]: Thank you for catching this!

**Review by tracychen1234 — APPROVED** [review]
Looking good!

File: com.unity.shadergraph/Editor/GraphDeltaRegistry/Registry/GradientType.cs [hunk 3]
```diff
@@ -76,10 +76,10 @@ public ShaderFoundry.ShaderFunction GetShaderFunction(NodeHandler node, ShaderFo
     /// </summary>
     internal class GradientNode : INodeDefinitionBuilder
     {
-        public RegistryKey GetRegistryKey() => new RegistryKey { Name = "GradientNode", Version = 1 };
+        public RegistryKey GetRegistryKey() => new RegistryKey { Name = "Gradient", Version = 1 };
         public RegistryFlags GetRegistryFlags() => RegistryFlags.Func;
 
-        public const string kInlineStatic = "InlineStatic";
+        public const string kInlineStatic = " ";
```
Comment (esmelusina) [inline com.unity.shadergraph/Editor/GraphDeltaRegistry/Registry/GradientType.cs:82]: @bencloward "GradientNode" and "InlineStatic" here are reference names in the generated code ("Gradient" is reserved and " " isn't a valid identifier). They are also used for display names right now, but we will be introducing another way to override those.

**Pull request merged**